src/app/courses/pages/new-course-page/new-course-page.component.ts

```diff
@@ -38,4 +64,8 @@ export class NewCoursPageComponent implements OnInit {
     });
   }
 
+  ngOnDestroy(): void {
```

Comment:
Lifecycle hooks should be at the beginning of the file and before other methods. Put ngOnDestroy after ngOnInit.